File: ai/src/main/java/org/conductoross/conductor/ai/models/LLMResponse.java
package org.conductoross.conductor.ai.models;

import java.util.List;

import com.netflix.conductor.common.metadata.workflow.WorkflowDef;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LLMResponse {
    private Object result;
    private List<Media> media;
    private String finishReason;
    private int tokenUsed;
    private int promptTokens;
    private int completionTokens;
    private List<ToolCall> toolCalls;
    private WorkflowDef workflow;
    private String jobId;
    private String responseId;
    private String reasoning;
    private Integer reasoningTokens;

    public boolean hasToolCalls() {
        return toolCalls != null && !toolCalls.isEmpty();
    }
}

File: ai/ai/src/models/LLMResponse.ts
import { Media } from './Media.js';
import { ToolCall } from './ToolCall.js';

export interface LLMResponse {
  result?: unknown;
  media?: Media[];
  finishReason?: string;
  tokenUsed?: number;
  promptTokens?: number;
  completionTokens?: number;
  toolCalls?: ToolCall[];
  jobId?: string;
  responseId?: string;
  reasoning?: string;
  reasoningTokens?: number;
}

export function hasToolCalls(response: LLMResponse): boolean {
  return (response.toolCalls?.length ?? 0) > 0;
}
